File: TapeEquilibrium/ruby/TapeEquilibrium.rb
class TapeEquilibrium
	def solution(a)

		right = 0
		left = 0
		result = 1000
		a.each { |a| right+=a }
		
		for i in 0...(a.count-1)

			left += a[i]
			right -= a[i]
			check = (left - right).abs
			result = check if (check < result)

		end

		result = (a[0]-a[1]).abs if a.count==2
		result
		
	end
end

File: TapeEquilibrium/php/TapeEquilibrium.php
<?php
class TapeEquilibrium {
	function solution($A) {

		$right = 0;
		$left = $A[0];
		$ar = array();

		for ($i = 1; $i<count($A); $i++) {
			$right += $A[$i];
		}


		for ($i = 1; $i<count($A); $i++) {

			$check = abs($left - $right);
			$ar[$i-1] = $check;
			$left += $A[$i];
			$right -= $A[$i];
		}

		$find = min($ar);

		if (count($A)==2) {

			$find = abs($A[0]-$A[1]);

		}

		return $find;

	}
}


